src/model/MoveGenerator.java

```diff
@@ -1,58 +1,70 @@
 
-    public MoveGenerator(Board board, PieceColor color) {
-        this.possibleMoves = new ArrayList<MoveInterface>();
+    public MoveGenerator(Board board) {
         this.board = board;
-        this.currentPlayersOccupiedSquares = new ArrayList<Square>();
-        this.determineThisPlayersOccupiedSquares(color);
     }
 
-    private void determineJumpMoves() {
-        for (Square startingSquare : this.currentPlayersOccupiedSquares) {
+    private Set<MoveInterface> calculateJumpMoves(Set<Square> playersSquares) {
+
+        Set<MoveInterface> possibleMoves = new HashSet<>();
+
+        for (Square startingSquare : playersSquares) {
             for (Square squareOneJumpAway : this.board
                     .getSquaresThatMightBeOneJumpAway(startingSquare)) {
+
                 SingleJump jump = new SingleJump(startingSquare.getPosition(),
                         squareOneJumpAway.getPosition(), this.board);
                 if (MoveValidator.isValidMove(jump)) {
-                    ArrayList<SingleJump> jumps = new ArrayList<SingleJump>();
+                    ArrayList<SingleJump> jumps = new ArrayList<>();
                     jumps.add(jump);
-                    ArrayList<Integer> intermediatePositions = new ArrayList<Integer>();
-                    intermediatePositions.add(jump.getEndingPosition());
-                    this.determineMultiJumpMoves(jumps, intermediatePositions);
+                    this.calculateMultiJumpMoves(jumps, possibleMoves);
                 }
             }
         }
+        return possibleMoves;
     }
 
-    private void determineMultiJumpMoves(ArrayList<SingleJump> jumps,
-            ArrayList<Integer> intermediatePositions) {
+    private void calculateMultiJumpMoves(ArrayList<SingleJump> jumps,
+            Set<MoveInterface> possibleMoves) {
+
         SingleJump lastJump = jumps.get(jumps.size() - 1);
 
         Board newBoard = new Board(lastJump.getBoard());
         newBoard.movePiece(lastJump);
 
-        Square lastSquare = lastJump.getEndingSquare();
+        boolean oldKingStatus = lastJump.getPiece().isKing();
+        boolean newKingStatus = newBoard.getPiece(lastJump.getEndingPosition()).isKing();
+        boolean wasKingingJump = oldKingStatus == false && newKingStatus == true;
 
         boolean noMoreJumps = true;
 
-        for (Square squareOneJumpAway : newBoard.getSquaresThatMightBeOneJumpAway(lastSquare)) {
-            SingleJump jump = new SingleJump(lastSquare.getPosition(),
-                    squareOneJumpAway.getPosition(), newBoard);
-            if (MoveValidator.isValidMove(jump)) {
-                noMoreJumps = false;
-                ArrayList<SingleJump> jumpsCopy = SingleJump.singleJumpListCopier(jumps);
-                jumpsCopy.add(jump);
-                intermediatePositions.add(jump.getEndingPosition());
-                this.determineMultiJumpMoves(jumpsCopy, intermediatePositions);
+        if (!wasKingingJump) { // must stop when kinged
```

Comment:
The wasKingingJump calculation in the move generator is used to stop the recursive multi-jumping since if a piece gets kinged, it must stop its turn. I'm sure determining if a singlejump was a kingingJump could be factored into another class, but it's not exactly a straightforward refactor and this gets this puppy working.